### PR TITLE
School Info Completeness:  Change from nces school to non nces school

### PR DESCRIPTION
**Pull request by nkiruka**
[Bug repro](https://docs.google.com/document/d/1KxMHcnQlTCKWJxHi0mK2uh_dBSIcqvWCe_4vTd7OCYI/edit#bookmark=id.s22r2awkgw6p)

Fix:  Clear the ncesSchoolId, when the country is changed.  
Result:  SchoolId changes from ncesSchoolId to null.  Since the schoolId changes, the update_and_add_users_school_infos is called and the database is updated with the new entry.

### CODE REVIEW

File: apps/test/unit/lib/ui/SchoolInfoInterstitialTest.jsx [hunk 1]
```diff
@@ -156,6 +157,43 @@ describe('SchoolInfoInterstitial', () => {
...
+            country: 'United States',
+            school_type: 'public',
+            school_name: 'Test School',
+            full_address: 'Seattle'
+          }
+        }}
+      />
+    );
+
+    expect(wrapper).to.containMatchingElement(
```
Comment (Hamms) [inline apps/test/unit/lib/ui/SchoolInfoInterstitialTest.jsx]: this should probably be in a different test

File: apps/test/unit/lib/ui/SchoolInfoInterstitialTest.jsx [hunk 2]
```diff
@@ -156,6 +157,43 @@ describe('SchoolInfoInterstitial', () => {
     );
   });
 
+  it('calls onChange function when country is changed from US to non-US', () => {
+    let wrapper = mount(
```
Comment (Hamms) [inline apps/test/unit/lib/ui/SchoolInfoInterstitialTest.jsx]: should this be
```suggestion
    const wrapper = shallow(
```

File: apps/test/unit/lib/ui/SchoolInfoInterstitialTest.jsx [hunk 3]
```diff
@@ -8,6 +8,7 @@ import Button from '@cdo/apps/templates/Button';
 import SchoolInfoInputs from '@cdo/apps/templates/SchoolInfoInputs';
 import SchoolInfoInterstitial from '@cdo/apps/lib/ui/SchoolInfoInterstitial';
 import firehoseClient from '@cdo/apps/lib/util/firehose';
+// import CountryAutocompleteDropdown from '@cdo/apps/templates/CountryAutocompleteDropdown';
```
Comment (Hamms) [inline apps/test/unit/lib/ui/SchoolInfoInterstitialTest.jsx]: this should probably be removed

File: apps/test/unit/lib/ui/SchoolInfoInterstitialTest.jsx [hunk 4]
```diff
@@ -156,6 +157,43 @@ describe('SchoolInfoInterstitial', () => {
     );
   });
 
+  it('calls onChange function when country is changed from US to non-US', () => {
```
Comment (Hamms) [inline apps/test/unit/lib/ui/SchoolInfoInterstitialTest.jsx]: this should probably be something like
```suggestion
  it('clears the school ID when country is changed', () => {
```

File: apps/test/unit/lib/ui/SchoolInfoInterstitialTest.jsx [hunk 5]
```diff
@@ -156,6 +157,43 @@ describe('SchoolInfoInterstitial', () => {
...
+        onCountryChange={wrapper.instance().onCountryChange}
+        onSchoolTypeChange={wrapper.instance().onSchoolTypeChange}
+        onSchoolChange={wrapper.instance().onSchoolChange}
+        onSchoolNotFoundChange={wrapper.instance().onSchoolNotFoundChange}
+      />
+    );
+    const event = {value: 'Sweden'};
+    wrapper.instance().onCountryChange(undefined, event);
+    expect(wrapper.state('country')).to.equal('Sweden');
+    expect(wrapper.state('ncesSchoolId')).to.equal('');
```
Comment (Hamms) [inline apps/test/unit/lib/ui/SchoolInfoInterstitialTest.jsx:179]: I'd love to see something like

```js
    expect(wrapper.state('country')).to.equal('United States');
    expect(wrapper.state('ncesSchoolId')).to.equal('123');
    wrapper.instance().onCountryChange(undefined, {value: 'Sweden'});
    expect(wrapper.state('country')).to.equal('Sweden');
    expect(wrapper.state('ncesSchoolId')).to.equal('');
```

So we can confirm the values actually _changed_

**Review by Hamms — APPROVED** [review]

File: apps/src/lib/ui/SchoolInfoInterstitial.jsx [hunk 6]
```diff
@@ -305,7 +305,7 @@ export default class SchoolInfoInterstitial extends React.Component {
 
   onCountryChange = (_, event) => {
     const newCountry = event ? event.value : '';
-    this.setState({country: newCountry, errors: {}});
+    this.setState({country: newCountry, ncesSchoolId: '', errors: {}});
```
Comment (bethanyaconnor) [inline apps/src/lib/ui/SchoolInfoInterstitial.jsx:308]: Should this be done on any other form changes?

Comment (nkiruka) [reply apps/src/lib/ui/SchoolInfoInterstitial.jsx:308]: Good question.  I will verify during the bug bash.

**Pull request merged**